ArduPlane/Plane.h

```diff
@@ -261,6 +264,9 @@ class Plane : public AP_HAL::HAL::Callbacks {
     // selected navigation controller
     AP_SpdHgtControl *SpdHgt_Controller = &TECS_controller;
 
+    //Soaring Controller
+    SoaringController soaring_controller {ahrs, SpdHgt_Controller, aparm, &DataFlash, LOG_THERMAL_MSG, LOG_VARIO_MSG};
```

Comment:
the DataFlash static thing is new since 3.2. Look at how other libraries do this, no one else passes the macro like this or needs the log format string.